spec/unit/datastream_verification_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe FedoraMigrate::DatastreamVerification do
+
+  class TestSubject
+    include FedoraMigrate::DatastreamVerification
+    def initialize datastream
+      @datastream = datastream
+    end
+  end
+
+  context "with a valid datastream" do
+    describe "checksums_match?" do
+      let(:mock_source) { double("Datastream", checksum: "foo") }
+      subject do
+        TestSubject.new(mock_source)
+      end
+      it "should return true for matching checksums" do
+        expect_any_instance_of(TestSubject).to receive(:target_checksum).once.and_return("foo")
+        expect(subject.checksums_match?).to be true
+      end
+    end
+
+    describe "no_checksum?" do
+      let(:mock_source) { double("Datastream", checksum: "none") }
+      subject do
+        TestSubject.new(mock_source)
+      end
+      it "should return true if Fedora has no checksum" do
+        expect(subject.no_checksum?).to be true
+      end
+    end
+
+    describe "content_equivalent?" do
+      let(:mock_source) { double("Datastream", checksum: "bad", mimeType: "text/xml", content: "<bar></bar>") }
+      subject do
+        TestSubject.new(mock_source)
+      end
+      it "should return true if xml content is equivalent" do
+        expect_any_instance_of(TestSubject).to receive(:target_content).once.and_return("<bar></bar>")
```

Comment:
This test would be better if the string passed into `#and_return` was equivalent XML but not an exact string match. Mimic the behavior that necessitates having an XML content check.